src/meta/src/hummock/manager/mod.rs

```diff
@@ -1109,136 +1154,253 @@ where
+                    .group_deltas;
+                group_deltas.push(GroupDelta {
                     delta_type: Some(DeltaType::GroupConstruct(GroupConstruct {
-                        group_config: Some(compaction_config),
+                        group_config: Some(compaction_config.clone()),
+                        parent_group_id: *parent_group_id,
+                        table_ids: Vec::from_iter(member_table_ids.iter().cloned()),
                     })),
                 });
+                let (non_trivial, split_id_vers) = new_hummock_version.init_with_parent_group(
+                    *parent_group_id,
+                    *group_id,
+                    member_table_ids,
+                );
+                if non_trivial {
+                    for (id, divide_ver) in split_id_vers {
+                        match branched_ssts.get_mut(id) {
+                            Some(mut entry) => {
+                                *entry.get_mut(parent_group_id).unwrap() += 1;
+                                entry.insert(*group_id, divide_ver);
+                            }
+                            None => branched_ssts.insert(
+                                id,
+                                [(*parent_group_id, divide_ver), (*group_id, divide_ver)]
+                                    .into_iter()
+                                    .collect(),
+                            ),
+                        }
+                    }
+                }
             }
         }
+
+        new_version_delta.max_committed_epoch = new_hummock_version.max_committed_epoch;
+        commit_multi_var!(self, None, new_version_delta)?;
+        branched_ssts.commit_memory();
+        versioning.current_version = new_hummock_version;
+
+        self.env
+            .notification_manager()
+            .notify_hummock_asynchronously(
+                Operation::Add,
+                Info::HummockVersionDeltas(risingwave_pb::hummock::HummockVersionDeltas {
+                    version_deltas: vec![versioning
+                        .hummock_version_deltas
+                        .last_key_value()
+                        .unwrap()
+                        .1
+                        .clone()],
+                    counterpart_compaction_groups: new_groups
+                        .iter()
+                        .map(|group_id| compaction_groups.get(group_id).unwrap().into())
+                        .collect_vec(),
+                    ..Default::default()
+                }),
+            );
+
+        Ok(None)
     }
 
     /// Caller should ensure `epoch` > `max_committed_epoch`
     #[named]
     pub async fn commit_epoch(
         &self,
         epoch: HummockEpoch,
-        sstables: Vec<LocalSstableInfo>,
+        mut sstables: Vec<LocalSstableInfo>,
         sst_to_context: HashMap<HummockSstableId, HummockContextId>,
     ) -> Result<()> {
+        let mut compaction_guard = write_lock!(self, compaction).await;
```

Comment:
Originally I thought this PR won't introduce overhead or regression if we don't trigger compaction split but this is not true. We need more careful reviews and benchmark before merging. I suggest we either revert this PR and re-merge it after we have ensured there is no performance regression or we prioritized refactoring this PR to address the potential issues. 

cc @soundOfDestiny @Little-Wallace @wenym1 @zwang28  What do you think?

Reply:
> We introduced more lock contention by acquiring the compaction lock in `commit_epoch`. For example, now `commit_epoch` competes with `get_compact_task_impl` and `assign_compaction_task`.
> 
> Why do we need compaction lock here? Is it because we need to modify `branched_ssts`. We need to be careful about that. We may be able to avoid this lock if we decouple sync/create group with commit epoch.

We cannot avoid this lock even if we decouple. Because `commit_epoch` can generate divided SSTs.

Reply:
> Originally I thought this PR won't introduce overhead or regression if we don't trigger compaction split but this is not true. We need more careful reviews and benchmark before merging. I suggest we either revert this PR and re-merge it after we have ensured there is no performance regression or we prioritized refactoring this PR to address the potential issues.
> 
> cc @soundOfDestiny @Little-Wallace @wenym1 @zwang28 What do you think?

I can repair them in #5888  in passing

Reply:
> We introduced more lock contention by acquiring the compaction lock in `commit_epoch`. For example, now `commit_epoch` competes with `get_compact_task_impl` and `assign_compaction_task`.
> 
> Why do we need compaction lock here? Is it because we need to modify `branched_ssts`. We need to be careful about that. We may be able to avoid this lock if we decouple sync/create group with commit epoch.

fixed in #5888